### PR TITLE
Define echeladder bonuses using enums

### PR DESCRIPTION
**Pull request by LunyaticCat**

### CODE REVIEW

File: src/main/java/com/mraof/minestuck/player/Echeladder.java [hunk 1]
```diff
@@ -65,9 +64,7 @@ public static void onPlayerRespawn(PlayerEvent.PlayerRespawnEvent event)
 	private final PlayerIdentifier identifier;
 	private int rung;
 	private int progress;
-	
-	private boolean[] underlingBonuses = new boolean[UNDERLING_BONUSES.length];
-	private boolean[] alchemyBonuses = new boolean[ALCHEMY_BONUSES.length];
+	EnumSet<EnumEcheladderBonuses> usedBonuses = EnumSet.noneOf(EnumEcheladderBonuses.class);
```
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/player/Echeladder.java:67]: Can be private final

**Review by kirderf1 — COMMENTED** [review]
Looks good. Just some style-related things.

File: src/main/java/com/mraof/minestuck/player/EnumEcheladderBonuses.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package com.mraof.minestuck.player;
+
+public enum EnumEcheladderBonuses
```
[the remark this answers is not in the document]
Comment (LunyaticCat) [reply src/main/java/com/mraof/minestuck/player/EnumEcheladderBonuses.java:3]: sure

**Review by kirderf1 — APPROVED** [review]
It's good to go, but if you'd like, you could write a javadoc for the new enum before I merge.

**Pull request merged**